### ISSUE TITLE
r command has trouble with prefixing.

### ISSUE THREAD

**Issue by jphartmann** [hercules-390/hyperion#72]
With prefix = x'4000'
```
HHC01603I abs 1000.8
HHC02290I A:0000000000001000  K:00
HHC02290I A:0000000000001000  00000000 00000000                    ........
```
However,  
```
HHC01603I r 1000.8
HHC02290I R:0000000000005000  K:F0
HHC02290I R:0000000000005000  00000000 00000000                    ........
```
I think the address printed should be the real address rather than the absolute one.  Of course, the data printed should be from the correct part of storage.

The problem is confusion in `hscmisc.c` function `dump_real_page `.   The real parameter is, in fact, the absolute address.  But perhaps it should not be since `dump_real_page` also applies prefixing.

Since my abs command throws another spanner in the works, I shall deal with it.

**Comment by jphartmann** [comment]
Fixed by commit `8b3cfe2e3e6093b853a1ece4a7581c7a299f536e`.  Also added ability to set the prefix register.

**Comment by Fish-Git** [comment]
FYI: Commit d32bf055eb0f6d30db9c4a58c8b582d6ce8a7de7 also applied further needed fixes.
